draft-ietf-ppm-dap.md

```diff
@@ -1200,34 +1182,120 @@ enum {
 } ReportShareError;
 ~~~
 
+The "initialize" and "continue" messages also include a payload, which is
+consumed by the VDAF. The `PrepareStep` is structured as follows:
+
+~~~
+enum {
+  initialize(0),
+  continue(1),
+  finished(2)
+  reject(3),
+  (255)
+} PrepareStepState;
+
+struct {
+  PrepareStepState prepare_step_state;
+  select (PrepareStep.prepare_step_state) {
+    case initialize:
+      ReportShare report_share;
+      opaque payload<0..2^32-1>;
+    case continue:
+      ReportId report_id;
+      opaque payload<0..2^32-1>;
+    case finished:
+      ReportId report_id;
+    case reject:
+      ReportId report_id;
+      ReportShareError report_share_error;
+  };
+} PrepareStep;
+~~~
+
+### Aggregate Initialization {#agg-init}
+
+The Leader begins an aggregation job by choosing a set of candidate reports that
+pertain to the same DAP task and a unique job ID. The job ID is a 16-byte value,
+structured as follows:
+
+~~~
+opaque AggregationJobID[16];
+~~~
+
+The Leader can run this process for many sets of candidate reports in parallel
+as needed. After choosing a set of candidates, the Leader begins aggregation by
+splitting each report into report shares, one for each Aggregator. The Leader
+and Helper then run the aggregate initialization flow to accomplish two tasks:
+
+1. Recover and determine which input report shares are valid.
+1. For each valid report share, initialize the VDAF preparation process (see
+   {{Section 5.2 of !VDAF}}).
+
 The Leader and Helper initialization behavior is detailed below.
 
 #### Leader Initialization {#leader-init}
 
 The Leader begins the aggregate initialization phase with the set of candidate
-report shares as follows:
+reports as follows:
 
 1. Generate a fresh AggregationJobID. The ID value MUST be unique within the
    scope of the corresponding DAP task.
```

Comment:
This requirement that the aggregation job ID be unique in the scope of the task seems correct, but doesn't quite align with what is said at the top of `{{agg-init}}`, which just says "unique", and I think that can only be read as "globally unique".

Anyway, the requirements for aggregation job ID should only be stated once, either here or at the top of `{{agg-init}}`.

Reply:
Bump, the inconsistent uniqueness requirements are still there at dcced627426a8db038b99c1c9918fe97de93299e

Reply:
Looking to https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/commit/dcced627426a8db038b99c1c9918fe97de93299e, `agg-init` now says "The Leader begins an aggregation job by choosing a set of candidate reports that
pertain to the same DAP task and a job ID which MUST be unique within the scope
of the task." while `leader-init` now only says "Generate a fresh AggregationJobID."

Reply:
Oops, never mind, I was holding it wrong.